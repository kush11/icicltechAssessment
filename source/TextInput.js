import React, { useState } from 'react';
import { View, TextInput, Text } from 'react-native';
import UserContext from '../context/card';
const TextInputs = ({ placeholder, value, Title }) => {
  const [getName, setName] = useState();
  const [getAge, setAge] = useState();
  const [getHobbies, setHobbies] = useState();
  const [getDescription, setDescription] = useState();
  return (
    <UserContext.Consumer>
      {context => (
        <View style={{ flex: 1, flexDirection: 'row' }}>
          <View style={{
            flex: 2,
            // borderWidth: 1, 
            justifyContent: 'center', alignItems: 'center'
          }}>
            <Text>{Title}:</Text>
          </View>
          <View style={{ flex: 7 }}>
            <TextInput
              style={{
                height: 40,
                margin: 5,
                borderBottomColor: 'gray',
                borderBottomWidth: 1
              }}
              placeholder={placeholder}
              // editable={false}              
              value={value ? value : null}
              onChangeText={
                (text) => {
                  if (placeholder === 'Enter your Age') {
                    setAge(text);
                    context.setUserAge(text);
                  }
                  if (placeholder === 'Enter your Hobby') {
                    setHobbies(text);
                    context.setUserHobbies(text);
                  }
                  if (placeholder === 'Write your shord Description about you') {
                    setDescription(text);
                    context.setUserDescription(text);
                  }
                  if (placeholder === 'Name') {
                    setName(text);
                    context.setUserName(text);
                  }

                }
              }
              keyboardType={placeholder === 'Enter your Age' ? 'number-pad' : 'default'}
            />
          </View>
        </View>
      )}
    </UserContext.Consumer>
  )
}

export default TextInputs;