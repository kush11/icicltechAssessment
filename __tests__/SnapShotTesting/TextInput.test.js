import 'react-native';
import React from 'react';
import TextInput from '../../source/TextInput';
import renderer from 'react-test-renderer';

test('TextInput snapshot ', () => {
    const snap = renderer.create(<TextInput/>).toJSON();
    expect(snap).toMatchSnapshot();
})

