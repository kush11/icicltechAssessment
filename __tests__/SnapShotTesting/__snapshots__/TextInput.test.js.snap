// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshot  1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      :
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 7,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      keyboardType="default"
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "borderBottomColor": "gray",
          "borderBottomWidth": 1,
          "height": 40,
          "margin": 5,
        }
      }
      underlineColorAndroid="transparent"
      value={null}
    />
  </View>
</View>
`;
